### ISSUE TITLE
False positive "not-callable" for "multiprocessing.current_process"

### ISSUE THREAD

**Issue by Delgan** [PyCQA/pylint#1699]
Hello.

### Steps to reproduce
Lint the following code:
```python
import multiprocessing
multiprocessing.current_process()
```

### Current behavior
This reports an issue: `E:  2, 0: multiprocessing.current_process is not callable (not-callable)`

### Expected behavior
No issue reported.

### pylint --version output
```
pylint 1.7.4,
astroid 1.5.3
Python 3.6.3 (default, Oct  4 2017, 02:55:45)
[GCC 5.4.0 20160609]
```

### Notes
Curiously, `threading.current_thread()` works fine.

**Comment by PCManticore** [comment]
This happens because we are inferring `current_process` as a staticmethod, coming from here: https://github.com/python/cpython/blob/master/Lib/multiprocessing/context.py#L37. We should check why the inference engine can't infer accessors such as this one as being callable. Th brain_multiprocessing plugin is here https://github.com/PyCQA/astroid/blob/9d94babb2e968c5a343218a48684a9529cbab568/astroid/brain/brain_multiprocessing.py, which just instantiates `BaseContext` and adds its members to the global namespace.
Basically when the issue gets fixed, the following should not emit `Uninferable` as it does now, but rather the `current_process` function:

```
from astroid import extract_node
n = extract_node('''
from multiprocessing.context import BaseContext
c = BaseContext()
c.current_process
''')
i = next(n.infer())
print(i)
```

**Comment by brycepg** [comment]
Here's what I found about `staticmethod`: a function wrapped with staticmethod isn't inferable unless it's defined in a class namespace (and called after the namespace is defined). Ex:

```python
def hi():
    pass
f = staticmethod(hi)
f # Not callable here
Class Foo:
    f = f
    print(f) # Still not callable
Foo.f() # callable
Foo().f() # callable
def bar():
    bar.f = f
bar.f # not callable
```

Astroid only infers staticmethod correctly using the `@staticmethod` syntax (using this decorator it correctly infers the attribute as a function instead of a bound or unbound method.

**Comment by bidiu** [comment]
Any update on this?

**Comment by brycepg** [comment]
After learning about python descriptors I'm thinking this is somewhat related to astroid not understanding them (PyCQA/astroid#118)

Until I get to that, I think I'll make an exception for `not-callable` with instances of staticmethod in pylint

**Comment by PCManticore** [comment]
@brycepg Sounds like a good plan to me.
